File: katas/8 kyu/counting-sheep-dot-dot-dot/solution.py
from itertools import count
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
count_sheeps = lambda x : x.count(True)
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def count_sheeps(arrayOfSheeps):
  return arrayOfSheeps.count(True)
# ----------------------------------- TEST ----------------------------------- #
array1 = [True,  True,  True,  False,
          True,  True,  True,  True ,
          True,  False, True,  False,
          True,  False, False, True ,
          True,  True,  True,  True ,
          False, False, True,  True ];
              
test.assert_equals(result := count_sheeps(array1), 17, "There are 17 sheeps in total, not %s" % result)

File: katas/8 kyu/counting-sheep-dot-dot-dot/solution.js
// counting-sheep-dot-dot-dot
/* ------------------------------- MY SOLUTION ------------------------------ */
function countSheeps(arrayOfSheep) {
  return arrayOfSheep.filter(Boolean).length
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
const chai = require("chai");
const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
chai.config.truncateThreshold=0;

describe("Tests", () => {
  it("test", () => {
var array1 = [true,  true,  true,  false,
              true,  true,  true,  true ,
              true,  false, true,  false,
              true,  false, false, true ,
              true,  true,  true,  true ,
              false, false, true,  true ];
              
Test(countSheeps(array1), 17, "There are 17 sheeps in total")
  });
});
